commit df507cde8d71063d5873a42f668156e4c32e86f9
Author: Patrick Donnelly <pdonnell@ibm.com>
Date:   Mon Dec 2 20:27:44 2019 -0800

    mgr: forward RADOS client instances for potential blacklist
    
    The mgr creates a per-module RADOS client connection for modules which
    interact with RADOS (e.g. the volumes module). These clients should also
    be blacklisted when the active mgr is failed; we don't want the former
    active mgr to continue interacting with RADOS when the new one takes
    over. This is particularly impactful for avoiding extraneous
    "unresponsive client" warnings from the MDS when the mgr switches
    (especially in testing). The MDS will pickup the new OSD blacklists
    which include's the old mgr's libcephfs instance and blacklist/evict
    that session quietly.
    
    Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

diff --git a/src/mgr/ActivePyModules.cc b/src/mgr/ActivePyModules.cc
index 4707a7d74d4..36a5f74bafe 100644
--- a/src/mgr/ActivePyModules.cc
+++ b/src/mgr/ActivePyModules.cc
@@ -1080,3 +1080,25 @@ void ActivePyModules::cluster_log(const std::string &channel, clog_type prio,
     clog->do_log(prio, message);
   }
 }
+
+void ActivePyModules::register_client(std::string_view name, std::string addrs)
+{
+  std::lock_guard l(lock);
+
+  entity_addrvec_t addrv;
+  addrv.parse(addrs.data());
+
+  dout(7) << "registering msgr client handle " << addrv << dendl;
+  py_module_registry.register_client(name, std::move(addrv));
+}
+
+void ActivePyModules::unregister_client(std::string_view name, std::string addrs)
+{
+  std::lock_guard l(lock);
+
+  entity_addrvec_t addrv;
+  addrv.parse(addrs.data());
+
+  dout(7) << "unregistering msgr client handle " << addrv << dendl;
+  py_module_registry.unregister_client(name, addrv);
+}
diff --git a/src/mgr/ActivePyModules.h b/src/mgr/ActivePyModules.h
index caa5915efa9..a8a0f3c4cf0 100644
--- a/src/mgr/ActivePyModules.h
+++ b/src/mgr/ActivePyModules.h
@@ -134,6 +134,9 @@ public:
   void clear_all_progress_events();
   void get_progress_events(std::map<std::string,ProgressEvent>* events);
 
+  void register_client(std::string_view name, std::string addrs);
+  void unregister_client(std::string_view name, std::string addrs);
+
   void config_notify();
 
   void set_uri(const std::string& module_name, const std::string &uri);
